skpro/regression/linear/_glm.py

```diff
@@ -182,11 +314,21 @@ def __init__(
         disp=False,
         max_start_irls=3,
         add_constant=False,
+        family="0",
```

Comment:
there is actually no condition that triggers the warning. The correct condition would be, "user passes a positional argument to `__init__`", but that is sth we cannot easily detect.

Unless you have an idea how to do that, I would suggest we use the final defaults already and not `"0"` etc, and we always raise the warning.